services/management/grpc/actions_server.go

```diff
@@ -83,24 +82,27 @@ func (s *actionsServer) StartPTSummaryAction(ctx context.Context, req *managemen
 }
 
 // StartPTMySQLSummaryAction starts pt-mysql-summary action.
-//nolint:lll,dupl
+//nolint:lll
 func (s *actionsServer) StartPTMySQLSummaryAction(ctx context.Context, req *managementpb.StartPTMySQLSummaryActionRequest) (*managementpb.StartPTMySQLSummaryActionResponse, error) {
-	ag, err := models.FindPMMAgentsForService(s.db.Querier, req.ServiceId)
+	// TODO https://jira.percona.com/browse/PMM-4172
+	// Accept node_id, not service_id.
```

Comment:
Why should we expect node_id instead of service_id?

Reply:
From https://www.percona.com/doc/percona-toolkit/3.0/pt-mysql-summary.html (and also from the linked ticket):

> The tool interacts minimally with the server upon which it runs. It assumes that you’ll run it on the same server you’re inspecting, and therefore it assumes that it will be able to find the my.cnf configuration file, for example. However, it should degrade gracefully if this is not the case. Note, however, that its output does not indicate which information comes from the MySQL database and which comes from the host operating system, so it is possible for confusing output to be generated if you run the tool on one server and connect to a MySQL database server running on another server.

It is not really a client-server Action as other Actions like EXPLAIN. It is much closer to the pt-summary.